classes/class-woothemes-sensei-certificates.php

```diff
@@ -1483,24 +1483,36 @@ public function enqueue_block_editor_assets() {
 	 * @return string Block HTML.
 	 */
 	public function update_view_certificate_button_url( $block_content, $block ): string {
+		$class_name = 'view-certificate';
+
 		// phpcs:ignore WordPress.Security.NonceVerification.Recommended -- Only used if the learner completed the course.
 		$course_id = isset( $_GET['course_id'] ) ? (int) $_GET['course_id'] : false;
 
-		// Check that the course ID exists and that the user has completed the course.
 		if (
+			// Check that the course ID exists and that the user has completed the course.
 			! $course_id
 			|| ! get_current_user_id()
 			|| 'course' !== get_post_type( $course_id )
 			|| ! Sensei_Utils::user_completed_course( $course_id, get_current_user_id() )
+
+			// Check that the block is a core/button and it contains the respective class name.
+			|| ! isset( $block['blockName'] )
+			|| 'core/button' !== $block['blockName']
+			|| ! isset( $block['attrs']['className'] )
+			|| false === strpos( $block['attrs']['className'], $class_name )
 		) {
 			return $block_content;
 		}
 
-		if ( ! method_exists( 'Sensei_Blocks', 'update_button_block_url' ) ) {
```

Comment:
I think it makes more sense to return an empty block in this case, so I changed this.